src/web/utils/FocusManager.ts

```diff
@@ -0,0 +1,198 @@
+/**
+* FocusManager.ts
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* Manages focusable elements for better keyboard navigation.
+*/
+
+import React = require('react');
+import ReactDOM = require('react-dom');
+import PropTypes = require('prop-types');
+import Types = require('../../common/Types');
+
+let _lastComponentId: number = 0;
+
+interface StoredFocusableComponent {
+    component: React.Component<any, any>;
+    restricted?: boolean;
+    origTabIndex?: number;
+}
+
+export class FocusManager {
+    private static rootFocusManager: FocusManager;
+    private static restrictionStack: FocusManager[] = [];
+    private static currentRestrictionOwner: FocusManager;
+
+    private static allFocusableComponents: {[id: string]: StoredFocusableComponent} = {};
```

Comment:
Opening curly brace should have space after it; similar for closing brace. Same comment below.